src/app/icd/client/RefreshKeySender.h

```diff
@@ -0,0 +1,137 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "ICDClientInfo.h"
+#include "ICDClientStorage.h"
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandSender.h>
+#include <app/OperationalSessionSetup.h>
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/ScopedNodeId.h>
+#include <lib/support/CodeUtils.h>
+#include <stddef.h>
+
+namespace chip {
+namespace app {
+
+class CheckInDelegate;
+/**
+ * @brief RefreshKeySender contains all the data and methods needed for key refresh and re-registration of an ICD client.
+ */
+class RefreshKeySender
+{
+public:
+    typedef Crypto::SensitiveDataBuffer<Crypto::kAES_CCM128_Key_Length> RefreshKeyBuffer;
+    /**
+     * @brief This class comprises command sender callbacks for the registration command.
+     */
+    class Callback : public CommandSender::Callback
+    {
+    public:
+        // CommandSender callbacks
+        void OnResponse(CommandSender * apCommandSender, const ConcreteCommandPath & aPath, const StatusIB & aStatus,
+                        chip::TLV::TLVReader * aData) override;
+        void OnError(const CommandSender * apCommandSender, CHIP_ERROR aError) override;
+        void OnDone(CommandSender * apCommandSender) override;
+
+        void SetRefreshKeySender(RefreshKeySender * apRefreshKeySender) { mpRefreshKeySender = apRefreshKeySender; }
+
+        virtual ~Callback() {}
+
+    private:
+        CHIP_ERROR mError                     = CHIP_NO_ERROR;
+        RefreshKeySender * mpRefreshKeySender = nullptr;
+    };
+
+    RefreshKeySender(CheckInDelegate * apCheckInDelegate, ICDClientInfo & aICDClientInfo, ICDClientStorage * aICDClientStorage);
+
+    /**
+     * @brief Used to send a re-registration command to the peer using a new key.
+     *
+     * @param[in] clientInfo    - ICDClientInfo object representing the state associated with the node that sent the check-in
+     *                            message. The callee can use the clientInfo to determine the type of key to generate.
+     * @param[in] keyData       - New key to re-register the client with the server
+     * @param[in] exchangeMgr   - exchange manager to use for the re-registration
+     * @param[in] sessionHandle - session handle to use for the re-registration
+     */
+    CHIP_ERROR RegisterClientWithNewKey(Messaging::ExchangeManager & exchangeMgr, const SessionHandle & sessionHandle);
+
+    /**
+     * @brief Sets up a CASE session to the peer for re-registering a client with the peer when a key refresh is required to avoid
+     * ICD counter rollover. Returns error if we did not even manage to kick off a CASE attempt.
+     */
+    CHIP_ERROR EstablishSessionToPeer();
+
+    /**
+     * @brief Getter for ICDClientInfo
```

Comment:
This is not a useful comment, honestly.  Just remove the comments that are obvious frim the functions.

And whatever getters here are not public API... please move them out of the public section.

Reply:
I don't need those anymore. Removed all of them.